politeiawww/cmsdatabase/testcmsdb/testcmsdb.go

```diff
@@ -0,0 +1,134 @@
+package testcmsdb
+
+import (
+	"sync"
+
+	cmsdb "github.com/decred/politeia/politeiawww/cmsdatabase"
+)
+
+type testcmsdb struct {
+	sync.RWMutex
+	shutdown bool // Backend is shutdown
+
+	records map[string]cmsdb.Invoice // [token]invoice
+}
+
+// ExchangeRate satisfies the db interface and is used in testing.
+func (c *testcmsdb) ExchangeRate(month, year int) (*cmsdb.ExchangeRate, error) {
+	// Values used in testing
+	return &cmsdb.ExchangeRate{
+		Month:        uint(month),
+		Year:         uint(year),
+		ExchangeRate: 1651,
+	}, nil
+}
+
+// InvoicesByAddress satisfies the db interface and is used in testing.
+func (c *testcmsdb) InvoicesByAddress(addr string) ([]cmsdb.Invoice, error) {
+	c.Lock()
```

Comment:
There's a difference between a read lock and a full lock.  When you only need to read data, only use a read lock.

```
c.RLock()
defer c.RUnlock()
```

When you need to read/write data, use the full lock.

```
c.Lock()
defer c.Unlock()
```

Same for below.

Reply:
Got it, thanks